spec/spec_helper_acceptance.rb

```diff
@@ -1,28 +1,58 @@
 
-unless ENV['RS_PROVISION'] == 'no'
-  install_puppet
-end
+if ENV['TARGET_HOST'].nil? || ENV['TARGET_HOST'] == 'localhost'
+  puts 'Running tests against this machine !'
+  if Gem.win_platform?
+    set :backend, :cmd
+  else
+    set :backend, :exec
+  end
+else
+  # load inventory
+  inventory_hash = inventory_hash_from_inventory_file
+  node_config = config_from_node(inventory_hash, ENV['TARGET_HOST'])
 
-RSpec.configure do |c|
-  # Readable test descriptions
-  c.formatter = :documentation
+  if target_in_group(inventory_hash, ENV['TARGET_HOST'], 'docker_nodes')
+    host = ENV['TARGET_HOST']
+    set :backend, :docker
+    set :docker_container, host
+  elsif target_in_group(inventory_hash, ENV['TARGET_HOST'], 'ssh_nodes')
+    set :backend, :ssh
+    options = Net::SSH::Config.for(host)
+    options[:user] = node_config.dig('ssh', 'user') unless node_config.dig('ssh', 'user').nil?
+    options[:port] = node_config.dig('ssh', 'port') unless node_config.dig('ssh', 'port').nil?
+    options[:keys] = node_config.dig('ssh', 'private-key') unless node_config.dig('ssh', 'private-key').nil?
+    options[:password] = node_config.dig('ssh', 'password') unless node_config.dig('ssh', 'password').nil?
+    options[:verify_host_key] = Net::SSH::Verifiers::Null.new unless node_config.dig('ssh', 'host-key-check').nil?
+    host = if ENV['TARGET_HOST'].include?(':')
+             ENV['TARGET_HOST'].split(':').first
+           else
+             ENV['TARGET_HOST']
+           end
+    set :host,        options[:host_name] || host
+    set :ssh_options, options
+    set :request_pty, true
+  elsif target_in_group(inventory_hash, ENV['TARGET_HOST'], 'winrm_nodes')
+    require 'winrm'
 
-  # Configure all nodes in nodeset
-  c.before :suite do
-    extend PuppetStrings::Acceptance::CommandUtils
-    hosts.each do |host|
-      scp_to(host, Dir.glob('puppet-strings*.gem').first, 'puppet-strings.gem')
-      install_ca_certs(host)
-      on host, "#{gem_command(host)} install yard"
-      on host, "#{gem_command(host)} install rgen"
-      on host, "#{gem_command(host)} install puppet-strings.gem"
+    set :backend, :winrm
+    set :os, family: 'windows'
+    user = node_config.dig('winrm', 'user') unless node_config.dig('winrm', 'user').nil?
+    pass = node_config.dig('winrm', 'password') unless node_config.dig('winrm', 'password').nil?
+    endpoint = "http://#{ENV['TARGET_HOST']}:5985/wsman"
 
-      scp_to(host, Dir.glob('spec/fixtures/acceptance/modules/test/pkg/username-test*.gz').first, 'test.tar.gz')
-      on host, puppet('module', 'install', 'test.tar.gz')
-    end
+    opts = {
+      user: user,
+      password: pass,
+      endpoint: endpoint,
+      operation_timeout: 300,
+    }
+
+    winrm = WinRM::Connection.new opts
+    Specinfra.configuration.winrm = winrm
   end
 end
+
+require 'spec_helper_acceptance_local' if File.file?(File.join(File.dirname(__FILE__), 'spec_helper_acceptance_local.rb'))
```

Comment:
any reason this is at the bottom of the file ?

Reply:
@glennsarti can you elaborate on why we need the `_local` version of `spec_helper_acceptance`? Does Litmus do something special with that? I don't think we are managing these files from a template with (e.g. `pdk update`) so it would seem fine to just have a few helpers directly in the main `spec_helper_acceptance`?

Reply:
> any reason this is at the bottom of the file ?
So that local `acceptance_local` can do configuration AFTER the target nodes have a transport ready for use, or override Litmus behaviour.  The default location (e.g. MOTD - https://github.com/puppetlabs/puppetlabs-motd/blob/055cd0668009f47a481f15cd10c041bd7641c8f9/spec/spec_helper_acceptance.rb#L3-L6) means you can't config nodes

Currently this functionality is not required, but could be in the future. Happy to move it up the top.

Reply:
>  can you elaborate on why we need the _local version of spec_helper_acceptance

Strictly speaking we don't. I was trying to keep this file as close as possible to what Litmus is expecting due to it being developed so quickly and making it easier to update in the future.  There was no consideration of `pdksync`